driver/vhci_ude/vhci_driver.c

```diff
@@ -14,21 +14,28 @@ cleanup_vhci(_In_ WDFOBJECT drvobj)
 	WPP_CLEANUP(WdfDriverWdmGetDriverObject((WDFDRIVER)drvobj));
 }
 
+extern PVOID vhci_pooltag_begin;
+
 static PAGEABLE VOID
 driver_unload(_In_ WDFDRIVER drvobj)
 {
-	UNREFERENCED_PARAMETER(drvobj);
-
+	PAGED_CODE();
 	TRD(DRIVER, "Enter");
+
+	ExFreePoolWithTag(vhci_pooltag_begin, VHCI_POOLTAG);
+	WPP_CLEANUP(WdfDriverWdmGetDriverObject((WDFDRIVER)drvobj));
 }
 
+DRIVER_INITIALIZE DriverEntry;
```

Comment:
Is it required?

Reply:
For building the driver this is not required, however when running code analyses over the driver this declaration fixes the following error:

Severity        Code    Description     Project File    Line    Suppression State
Warning C28101  This function appears to be an unannotated DriverEntry function:  DriverEntry functions should be declared using the DRIVER_INITIALIZE function typedef.        usbip_vhci_ude  C:\Users\samo\Source\Repos\usbip-win\driver\vhci_ude\vhci_driver.c      32